lib/sync_dao.es6.js

```diff
@@ -93,22 +95,29 @@ foam.CLASS({
       factory: function() {
         this.validate();
 
+        // TODO(markdittmer): Clean this up. Need a way to override lookup and
```

Comment:
@adamvy any insight on how to build axioms for the right pattern here? What I'm manually doing is the following:

- This instance may have been created in a whitelisted context. Now that deserialization is complete, it wishes to instantiate not-whitelisted-for-serialization objects.
- It does this by using another context (that contains important info, but isn't whitelist-managed) to perform `lookup()` (see `this.lookup()` implementation), and
- Passing in the context to each `create()` call

This gets the job done, but it's really verbose. Also, this class `requires`ing classes does not guarantee that the "other context I want to use for lookup and creation" knows about those classes. (It happens to work in this case, but such a guarantee would be nice.)

Any thoughts on how to reify this into a common pattern? I think it will come up again as more and more as server-side apps need isolated contexts.